src/components/Message/MessageSimple/MessageContent.js

```diff
@@ -93,683 +82,592 @@ const FailedText = styled.Text`
-     * Defaults to: https://github.com/GetStream/stream-chat-react-native/blob/master/src/components/ReactionList.js
-     */
-    ReactionList: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
-    /** Boolean - if reaction picker is visible. Hides the reaction list in that case */
-    reactionPickerVisible: PropTypes.bool,
-    /** enabled reactions, this is usually set by the parent component based on channel configs */
-    reactionsEnabled: PropTypes.bool.isRequired,
-    /**
-     * @deprecated Please use `disabled` instead.
-     *
-     * Disables the message UI. Which means, message actions, reactions won't work.
-     */
-    readOnly: PropTypes.bool,
-    /** enabled replies, this is usually set by the parent component based on channel configs */
-    repliesEnabled: PropTypes.bool.isRequired,
-    /**
-     * e.g.,
-     * [
-     *  {
-     *    id: 'like',
-     *    icon: '👍',
-     *  },
-     *  {
-     *    id: 'love',
-     *    icon: '❤️️',
-     *  },
-     *  {
-     *    id: 'haha',
-     *    icon: '😂',
-     *  },
-     *  {
-     *    id: 'wow',
-     *    icon: '😮',
-     *  },
-     * ]
-     */
-    supportedReactions: PropTypes.array,
-    /**
-     * Custom UI component to display enriched url preview.
-     * Defaults to https://github.com/GetStream/stream-chat-react-native/blob/master/src/components/Card.js
-     */
-    UrlPreview: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
-  };
-
-  static defaultProps = {
-    Attachment,
-    FileAttachment,
-    FileAttachmentGroup,
-    Gallery,
-    hideReactionCount: false,
-    hideReactionOwners: false,
-    MessageReplies,
-    MessageText: false,
-    ReactionList,
-    reactionsEnabled: true,
-    repliesEnabled: true,
-    supportedReactions: emojiData,
-  };
-
-  constructor(props) {
-    super(props);
-
-    this.ActionSheet = false;
-    this.state = {};
-  }
-
-  openThread = () => {
-    if (this.props.onThreadSelect)
-      this.props.onThreadSelect(this.props.message);
-  };
-
-  showActionSheet = async () => {
-    await this.props.dismissKeyboard();
-    this.ActionSheet.show();
-  };
-
-  handleDelete = async () => {
-    await this.props.handleDelete();
-  };
-
-  handleEdit = () => {
-    this.props.handleEdit();
-  };
-
-  /**
-   * @todo: Remove the method in future 1.0.0.
-   * This method has been moved to `ReactionPickerWrapper`.
-   *
-   * @deprecated
-   */
-  _setReactionPickerPosition = async () => {
-    console.warn(
-      'openReactionSelector has been deprecated and will be removed in next major release.' +
-        'Please use this.props.openReactionPicker instead.',
-    );
-
-    await this.props.openReactionPicker();
+/**
+ * Since this component doesn't consume `messages` from `MessagesContext`,
+ * we memoized and broke it up to prevent new messages from re-rendering
+ * each individual MessageContent component.
+ */
+const MessageContentWithContext = React.memo((props) => {
+  const {
```

Comment:
works for me, will update

Reply:
I started moving some of these around but it started to make the file pretty messy. One of my biggest hurdles in jumping into the first version of the library was figuring out what props we were using in the component since the destructuring was happening in many different places. I think for now it'll be cleanest to keep them all together, but maybe we can re-think the hierarchy of the Message components and break them down a bit further to lessen some of these files

Reply:
agreed